istioctl/pkg/verifier/verifier.go

```diff
@@ -106,6 +115,26 @@ func (v *StatusVerifier) verifyInstallIOPRevision() error {
 	return v.reportStatus(crdCount, istioDeploymentCount, err)
 }
 
+func getRevision(istioNamespace, kubeConfig, ctx string) (string, error) {
+	var revision string
+	cfg, err := kube.BuildClientConfig(kubeConfig, ctx)
+	if err != nil {
+		return "", err
+	}
+	kubeClient, err := kube.NewExtendedClient(kube.NewClientConfigForRestConfig(cfg), "")
+	if err != nil {
+		return "", fmt.Errorf("failed to connect Kubernetes API server, error: %v", err)
+	}
```

Comment:
Is there a common function to init the kubeClient?

Reply:
I'm worried about the same, there has to be an easier way to initialize `kubeClient`, this block of code looks like a lot of work just to find revision.

Reply:
Pretty much everywhere we call `istioKube.NewExtendedClient(istioKube.NewClientConfigForRestConfig(rc), "")` or some variation.

Originally _istioctl_ and Pilot had different kube clients.  The code was refactored to a single client by a Pilot expert who made it work but did not focus on making it pretty for _istioctl_.  Probably it would be easier to refactor if the Kube context wasn't a package-level variable -- that was a mistake.